crates/matrix-sdk-ui/src/timeline/controller/state.rs

```diff
@@ -342,9 +346,33 @@ impl TimelineStateTransaction<'_> {
 
         self.adjust_day_dividers(day_divider_adjuster);
 
+        self.check_no_unused_unique_ids();
         total
     }
 
+    fn check_no_unused_unique_ids(&self) {
+        let mut known_unique_ids = HashSet::new();
```

Comment:
We depend on `itertools` so we can do this instead:

```rust
let duplicates: Vec<_> = self.items.iter().duplicates_by(|item| item.unique_id()).collect();
```

More info: https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.duplicates_by.

Reply:
Indeed. `duplicates_by` returns the `Arc<TimelineItem>`, so I added a `map` to get the unique id itself instead.